api/middleware.go

```diff
@@ -0,0 +1,78 @@
+package api
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+// ManagerURLMiddleware injects X-Forwarded-* headers based on manager-url setting.
+// This middleware reads the manager-url setting and, when configured, overrides
+// the X-Forwarded-Proto, X-Forwarded-Host, and X-Forwarded-Port headers to ensure
+// API responses (actions and links fields) contain the correct external URL instead
+// of internal pod IPs.
+// Note: The manager-url value is validated when the setting is updated, so no
+// validation is performed here.
+func ManagerURLMiddleware(s *Server) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			managerURL, err := s.m.GetSettingValueExisted(types.SettingNameManagerURL)
+			if err != nil {
+				logrus.WithError(err).Warn("Failed to get manager-url setting, using default behavior")
```

Comment:
Can we change the `Warn` to `Trace`?
If the value is empty, there will be lots of warning messages.